static/js/tasks_browse.js

```diff
@@ -536,23 +541,39 @@ function displayTaskInfo(taskInfo, data) {
         return;
     }
 
+    var dataInfo = data["taskruns_info"]
     var headers = [];
     var info = '<table class="table"><thead><tr>';
     // Extract headers
-    for (var headerName in data[0].info) {
+    for (var headerName in dataInfo[0].info) {
         headers.push(headerName);
         info += '<th>' + headerName + '</th>'
     }
     info += '</tr></thead><tbody>';
-    for (var i = 0; i < data.length; i++) {
+    for (var i = 0; i < dataInfo.length; i++) {
         info += '<tr>';
         for (var j = 0; j < headers.length; j++) {
             var headerName = headers[j];
-            info += '<td>' + JSON.stringify(data[i].info[headerName]) + '</td>';
+            info += '<td>' + JSON.stringify(dataInfo[i].info[headerName]) + '</td>';
         }
         info += '</tr>';
     }
     info += '</tbody></table>';
+
+    var dataGoldAnswers = data["gold_answers"]
+    if (!Object.keys(dataGoldAnswers).length){
+        taskInfo.html(info);
```

Comment:
can add the gold info if `dataGoldAnswers` is populated and call `taskInfo.html` once outside the `if` body.